nullaway/src/main/java/com/uber/nullaway/handlers/OptionalEmptinessHandler.java

```diff
@@ -155,33 +181,81 @@ private void updateIfAssertIsPresentTrueOnOptional(
             MethodInvocationNode argMethod = (MethodInvocationNode) unwrappedArg;
             Symbol.MethodSymbol argSymbol = ASTHelpers.getSymbol(argMethod.getTree());
             if (optionalIsPresentCall(argSymbol, types)) {
-              updateNonNullAPsForElement(
-                  bothUpdates, getOptionalGetter(argSymbol), argMethod.getTarget().getReceiver());
+              updateNonNullAPsForOptionalContent(bothUpdates, argMethod.getTarget().getReceiver());
             }
           }
         }
       }
     }
   }
 
-  private void updateNonNullAPsForElement(
-      AccessPathNullnessPropagation.Updates updates, @Nullable Element elem, Node base) {
-    if (elem != null) {
-      AccessPath ap = AccessPath.fromBaseAndElement(base, elem);
-      if (ap != null) {
-        updates.set(ap, Nullness.NONNULL);
-      }
+  private void updateNonNullAPsForOptionalContent(
+      AccessPathNullnessPropagation.Updates updates, Node base) {
+    AccessPath ap = AccessPath.fromBaseAndElement(base, getOptionalContentElement());
+    if (ap != null && base.getTree() != null) {
+      optionalExpressionToAccessPathMap.put(base.getTree().toString(), ap);
+      updates.set(ap, Nullness.NONNULL);
     }
   }
 
-  private Element getOptionalGetter(Symbol.MethodSymbol symbol) {
-    for (Symbol elem : symbol.owner.getEnclosedElements()) {
-      if (elem.getKind().equals(ElementKind.METHOD)
-          && elem.getSimpleName().toString().equals("get")) {
-        return elem;
+  private VariableElement getOptionalContentElement() {
+    return new VariableElement() {
+
+      @Override
+      public Object getConstantValue() {
+        return null;
       }
-    }
-    return null;
+
+      @Override
+      public Name getSimpleName() {
+        return StringName.of("OPTIONAL_CONTENT");
```

Comment:
Rather than doing string comparisons on `"OPTIONAL_CONTENT"`, we should keep a single `VariableElement` for optional content in a static field, and do comparisons based on object equality with that constant.  What do you think?